### PR TITLE
Uplink setup: respect tls args in libuplink config

### PR DESCRIPTION
**Pull request by bryanchriswhite**
What: Use tls cli args to populate libuplink config in uplink setup.

Why: The libuplink calls in uplink setup were passing a `nil` uplink config, as a result tls cli arguments weren't totally respected.

Please describe the tests:

To test for this one can run `uplink setup` with an uplink binary and try to point it at a storj-sim instance. When running with a binary built prior to this:
```
$ uplink setup --tls.use-peer-ca-whitelist=false
Select your satellite:
        [1] us-central-1.tardigrade.io
        [2] europe-west-1.tardigrade.io
        [3] asia-east-1.tardigrade.io
Please enter numeric choice or enter satellite address manually [1]: 2019-08-08T12:18:19.174+0200       DEBUG   process/debug.go:51      debug server listening on 127.0.0.1:38279
localhost:10000
Enter your API key: 13YqdperyLuzWbYcGZKBzuwRrAZi9iUspk55xuzUHfguRuz6bboEKw9LgEU4Ej8NSSkZzE8FL9595Qtj6EKwgjujEE1e9pM9vhYe5M2
Enter your encryption passphrase: 
Enter your encryption passphrase again: 
2019-08-08T12:18:38.102+0200    FATAL   process/exec_conf.go:286        uplink: metainfo error: transport error: connection error: desc = "transport: authentication handshake failed: tls peer certificate verification error: not signed by any CA in the whitelist: CA cert"
storj.io/storj/pkg/process.cleanup.func1
        /home/bwhite/Projects/new-storj/pkg/process/exec_conf.go:286
github.com/spf13/cobra.(*Command).execute
        /home/bwhite/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:762
github.com/spf13/cobra.(*Command).ExecuteC
        /home/bwhite/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852
github.com/spf13/cobra.(*Command).Execute
        /home/bwhite/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
storj.io/storj/pkg/process.Exec
        /home/bwhite/Projects/new-storj/pkg/process/exec_conf.go:72
main.main
        /home/bwhite/Projects/new-storj/cmd/uplink/main.go:12
runtime.main
        /usr/local/go/src/runtime/proc.go:200
```
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: cmd/uplink/cmd/setup.go [hunk 1]
```diff
@@ -120,7 +120,11 @@ func cmdSetupInteractive(cmd *cobra.Command, setupDir string) error {
 		return Error.Wrap(err)
 	}
 
-	uplink, err := libuplink.NewUplink(ctx, nil)
+	uplinkCfg := libuplink.Config{}
+	uplinkCfg.Volatile.TLS.PeerCAWhitelistPath = vip.GetString("tls.peer-ca-whitelist-path")
+	uplinkCfg.Volatile.TLS.SkipPeerCAWhitelist = !vip.GetBool("tls.use-peer-ca-whitelist")
```
Comment (egonelbre) [inline cmd/uplink/cmd/setup.go]: Are there other arguments that should be supplied as well, not just TLS?

Doesn't gateway have the same issue?

Comment (bryanchriswhite) [reply cmd/uplink/cmd/setup.go]: Looks like the gateway already does this: https://github.com/storj/storj/blob/master/cmd/gateway/main.go#L244

Comment (bryanchriswhite) [reply cmd/uplink/cmd/setup.go]: And yea, there are other uplink config fields that aren't being set but I don't think the existing bound config includes the others.

Comment (kaloyan-raev) [reply cmd/uplink/cmd/setup.go]: Note that we have this helper method: https://github.com/storj/storj/blob/master/cmd/uplink/cmd/root.go#L77-L91

Comment (bryanchriswhite) [reply cmd/uplink/cmd/setup.go]: @kaloyan-raev that's cool, I was not aware. We'd have reorganize the config structs i think to use that though.

**Pull request merged**